File: Find the duplicate number/find the duplicate number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        start = 1
        end = len(nums) - 1
        
        ans = 1
        while (start <= end):
            mid = (start + end)//2
            
            count = 0
            for num in nums:
                if (num <= mid):
                    count += 1
            
            if (count <= mid):
                start = mid + 1
            else:
                ans = mid
                end = mid - 1
        
        return ans

File: Find the duplicate number/find the duplicate number.java
class Solution {
    public int findDuplicate(int[] nums) {
        int start = 1;
        int end = nums.length - 1;
        
        int ans = 1;
        
        while (start <= end) {
            int mid = (start + end)/2;
            
            int count = 0;
            for(int i = 0; i < nums.length; i++) {
                if (nums[i] <= mid) {
                    count++;
                }
            }
            
            if (count <= mid) {
                start = mid + 1;
            }
            else {
                ans = mid;
                end = mid - 1;
            }
            
        }
        
        return ans;
    }
}